src/crimson/common/gated.h

```diff
@@ -52,4 +68,43 @@ class Gated {
   seastar::gate pending_dispatch;
 };
 
-}// namespace crimson::common
+class gate_per_shard : public Gated {
```

Comment:
Can you please add a comment for future readers explaining why this concept was introduced?

Reply:
Done